### PR TITLE
nixos/wireless: enable PMF by default

### PR DESCRIPTION
**Pull request by rnhmjoj**
###### Motivation for this change

Alternative solution to PR #152443.
This fixes authentication failures to WPA3 networks (issue #151729)
by enabling protected management frames.

Note: old client without 802.11w support will still fail.

###### Things done

- [x] Tested connecting to SAE, Mixed WPA-PSK/SAE network
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by rnhmjoj** [comment]
@mweinelt @blocktrron

**Comment by mweinelt** [comment]
I'm confused.

Is this blanket breakage of older stations, or will they continue working fine with networks that don't require it? Setting `pmf=1` unconditionally in there is a difficult spot for a user to revert that change locally fwiw.

**Comment by rnhmjoj** [comment]
> Is this blanket breakage of older stations, or will they continue working fine with networks that don't require it? Setting pmf=1 unconditionally in there is a difficult spot for a user to revert that change locally fwiw.

No, the only consequence of `pmf=1` is allowing connections to networks that require 802.11w.
Older clients will still fail on WPA3 networks, but work normally otherwise. At least, this is what I'm seeing from testing with a bunch of devices.

**Comment by rnhmjoj** [comment]
I think I've found a solution to keep backward compatibility without sacrificing security too much.

File: nixos/modules/services/networking/wpa_supplicant.nix [hunk 1]
```diff
@@ -175,6 +202,18 @@ in {
         '';
       };
 
+      opportunisticWPA3 = mkOption {
```
[the remark this answers is not in the document]
Comment (rnhmjoj) [reply nixos/modules/services/networking/wpa_supplicant.nix:205]: Yeah, good idea.

**Comment by tfc** [comment]
I just checked this branch out on my Lenovo x250 and i can confirm now that this **fixes** my issue (https://github.com/NixOS/nixpkgs/issues/151729).

**Review by tfc — APPROVED** [review]

File: nixos/modules/services/networking/wpa_supplicant.nix (hunk 1 above)
[the remark this answers is not in the document]
Comment (rnhmjoj) [reply nixos/modules/services/networking/wpa_supplicant.nix]: Name updated!

**Comment by rnhmjoj** [comment]
I believe @tfc tested this with modern hardware with PMF. It would be good if someone other than me checked the fallback works by running this on an old network card.

**Comment by rnhmjoj** [comment]
It's been a while but I haven't managed to get anyone else to test this change.
I'm merging it now anyway because I don't want to leave WPA3 (completely) broken any longer.

**Pull request merged**